rpcs3/Emu/system_progress.hpp

```diff
@@ -38,10 +38,11 @@ class scoped_progress_dialog final
 	scoped_progress_dialog(const std::string& text) noexcept
 	{
 		ensure(!text.empty());
-		m_text = text;
 
-		std::tie(m_prev, m_prev_id, m_id) = g_progr_text.atomic_op([this](progress_dialog_string_t& progr)
+		std::tie(m_prev, m_prev_id, m_id) = g_progr_text.atomic_op([this, text](progress_dialog_string_t& progr)
 		{
+			m_text = text;
```

Comment:
There is no race in access of data inside a constructor, a constructor (should) occur only before the object the is accessable to everyone else.

Reply:
true